http.go

```diff
@@ -22,73 +22,75 @@ import (
+		var wantedIP uint32
+		var wantedPV uint32
+		var wantedSF wire.ServiceFlag
 
-	query := r.URL.Query()
+		query := r.URL.Query()
 
-	requestedIP := query.Get(api.IPVersion)
-	if requestedIP != "" {
-		u, _ := strconv.ParseUint(requestedIP, 10, 32)
-		if u == 4 || u == 6 {
-			wantedIP = uint32(u)
+		requestedIP := query.Get(api.IPVersion)
+		if requestedIP != "" {
+			u, _ := strconv.ParseUint(requestedIP, 10, 32)
+			if u == 4 || u == 6 {
+				wantedIP = uint32(u)
+			}
 		}
-	}
-
-	requestedPV := query.Get(api.ProtocolVersion)
-	if requestedPV != "" {
-		u, _ := strconv.ParseUint(requestedPV, 10, 32)
-		wantedPV = uint32(u)
-	}
-
-	requestedSF := query.Get(api.ServiceFlag)
-	if requestedSF != "" {
-		u, _ := strconv.ParseUint(requestedSF, 10, 64)
-		wantedSF = wire.ServiceFlag(u)
-	}
 
-	nodes := amgr.GoodAddresses(wantedIP, wantedPV, wantedSF)
-
-	flush, ok := w.(http.Flusher)
-	if !ok {
-		http.NotFound(w, r)
-		return
-	}
+		requestedPV := query.Get(api.ProtocolVersion)
+		if requestedPV != "" {
+			u, _ := strconv.ParseUint(requestedPV, 10, 32)
+			wantedPV = uint32(u)
+		}
 
-	w.Header().Set("Content-Type", "text/plain; charset=utf-8") // not a json array
-	// Replace the Server response header. When used with nginx's "server_tokens
-	// off;" and "proxy_pass_header Server;" options.
-	w.Header().Set("Server", appName)
-	w.WriteHeader(http.StatusOK)
-	flush.Flush()
+		requestedSF := query.Get(api.ServiceFlag)
+		if requestedSF != "" {
+			u, _ := strconv.ParseUint(requestedSF, 10, 64)
+			wantedSF = wire.ServiceFlag(u)
+		}
 
-	enc := json.NewEncoder(w)
+		nodes := amgr.GoodAddresses(wantedIP, wantedPV, wantedSF)
 
-	ctx := r.Context()
-	for _, node := range nodes {
-		select {
-		case <-ctx.Done():
+		flush, ok := w.(http.Flusher)
+		if !ok {
+			http.NotFound(w, r)
 			return
-		default:
-			err := enc.Encode(node)
-			if err != nil {
-				log.Printf("httpGetAddrs: Encode failed: %v", err)
+		}
+
+		w.Header().Set("Content-Type", "text/plain; charset=utf-8") // not a json array
+		// Replace the Server response header. When used with nginx's "server_tokens
+		// off;" and "proxy_pass_header Server;" options.
+		w.Header().Set("Server", appName)
+		w.WriteHeader(http.StatusOK)
+		flush.Flush()
+
+		enc := json.NewEncoder(w)
+
+		ctx := r.Context()
+		for _, node := range nodes {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				err := enc.Encode(node)
+				if err != nil {
+					log.Printf("httpGetAddrs: Encode failed: %v", err)
+				}
+				flush.Flush()
 			}
-			flush.Flush()
 		}
 	}
 }
 
-func serveHTTP(ctx context.Context, addr string) error {
+func serveHTTP(ctx context.Context, addr string, amgr *Manager) error {
 	listener, err := net.Listen("tcp", addr)
 	if err != nil {
 		return fmt.Errorf("can't listen on %s. web server quitting: %w", addr, err)
 	}
 
 	mux := http.NewServeMux()
-	mux.HandleFunc(api.GetAddrsPath, httpGetAddrs)
+	mux.HandleFunc(api.GetAddrsPath, httpGetAddrs(amgr))
```

Comment:
Not a big deal, but this would result in a much smaller diff that is easier to review doing it like so:

```suggestion
	mux.HandleFunc(api.GetAddrsPath, func(w http.ResponseWriter, r *http.Request) {
		httpGetAddrs(amgr, w, r)
	})
```

```diff
-func httpGetAddrs(w http.ResponseWriter, r *http.Request) {
+func httpGetAddrs(amgr *Manager, w http.ResponseWriter, r *http.Request) {
```

Reply:
I like it, changed.